### ISSUE TITLE
Unsupported string offset operations false-positive on array assignment

### ISSUE THREAD

**Issue by fredricj** [kalessil/phpinspectionsea#824]
The following (simplified) code triggers "Unsupported string offset operations: Provokes a PHP Fatal error (cannot use string offset as an array)":
```<?php
$row = [1,2,3];
$plates = [];
foreach ($row as $t) {
	$header = "t";
	$plates[$header][] = $t; // Gives "Cannot use string offset as an array"
}
```

**Comment by kalessil** [comment]
Thank you for reporting @fredricj. For some reason, I can not reproduce. Would you please check this code fragments?

```php
function local_scope_only() {
    $row = [1,2,3];
    $plates = [];
    foreach ($row as $t) {
	    $header = "t";
	    $plates[$header][] = $t; // Gives "Cannot use string offset as an array"
    }
}
```

**Comment by fredricj** [comment]
Wrapping it in a function does not trigger the message but removing the function scope from your code fragment emits the message again

**Comment by kalessil** [comment]
Got it: I'll make adjustments. The inspection will skip global context and stick to functions and methods only.

**Comment by kalessil** [comment]
Fixed
